physics/cires_ugwp_v1.F90

```diff
@@ -0,0 +1,635 @@
+
+    use cires_ugwp_module_v1, only: knob_ugwp_version, cires_ugwp_init_v1, &
+                                 cires_ugwp_finalize_v1
+
+    use gwdps, only: gwdps_run
+
+    implicit none
+
+    private
+
+    public cires_ugwp_v1_run
+
+    logical :: is_initialized = .False.
+
+
+contains
+
+
+
+! -----------------------------------------------------------------------
+! CCPP entry points for CIRES Unified Gravity Wave Physics (UGWP) scheme v0
+! -----------------------------------------------------------------------
+!    originally from cires_ugwp_driver_v1.f
+!    driver of cires_ugwp   (_driver)
+! -----------------------------------------------------------------------
+!   driver is called after pbl & before chem-parameterizations
+! -----------------------------------------------------------------------
+!  order = dry-adj=>conv=mp-aero=>radiation -sfc/land- chem -> vertdiff-> [rf-gws]=> ion-re
+! -----------------------------------------------------------------------
+!>@brief The subroutine initializes the CIRES UGWP
+!> \section arg_table_cires_ugwp_init Argument Table
+!! \htmlinclude cires_ugwp_v1_init.html
+!!
+! -----------------------------------------------------------------------
+!
+
+
+subroutine cires_ugwp_v1_init(Model,Init_parm,Grid)
+                ! NOTE:  This code is taken from Valery's code in subroutine
+                ! GFS_initialize in GFS_driver.F90
+
+use cires_ugwp_module,   only: cires_ugwp_init_emc, cires_ugwp_init_v1, cires_indx_ugwp
+
+implicit none
+
+real(kind=kind_phys), parameter   :: p_ref = 101325.0d0
+
+
+!----  initialization of cires_ugwp .
+!     if ( Model%me == Model%master) print *,  ' VAY-nml ',  Model%fn_nml
+!     if ( Model%me == Model%master) print *,  ' VAY-nml2 ', Model%input_nml_file
+    if (Model%cdmbgwd(3) > 0.0) then 
+!     if ( Model%me == Model%master) print *,  ' VAY-nml ',  Model%fn_nml,
+!     Model%input_nml_file
+     call cires_ugwp_init_emc(Model%me,      Model%master, Model%nlunit,  Init_parm%logunit, &
+                           Model%fn_nml,  Model%lonr,   Model%latr,    Model%levs,          &    
+                           Init_parm%ak,  Init_parm%bk, p_ref,         Model%dtp,           &    
+                           Model%cdmbgwd(1:4), Model%cgwf,   Model%prslrd0, Model%ral_ts)
+                           
+    endif
+
+   if (Model%do_ugwp .and. Model%cdmbgwd(3) == 0.0 ) then 
+!
+
+    
+     call cires_ugwp_init_v1(Model%me,      Model%master, Model%nlunit,  Init_parm%logunit, &
+                Model%Jdat, Model%fn_nml,  Model%lonr,   Model%latr,    Model%levs,         &    
+                            Init_parm%ak,  Init_parm%bk, p_ref,         Model%dtp,          &    
+                           Model%cdmbgwd(1:2), Model%cgwf,   Model%prslrd0, Model%ral_ts)  
+                           
+!                          
+
+     do nb = 1, nblks
+     
+       call cires_indx_ugwp (Init_parm%blksz(nb),Model%me,  Model%master, Grid(nb)%xlat_d,  &
```

Comment:
Note that the init routine will not receive blocked data, but one contiguous array for all blocks of an MPI task.